src/autocomplete/autocomplete.ts

```diff
@@ -169,7 +176,7 @@ export function parseQuery<
     generateSuggestionsFromRules: GenerateSuggestionsFromRules<A>,
     query: string,
     cursor: CursorPosition,
-): AutocompleteResultBase {
+): MySqlAutocompleteResult & PostgreSqlAutocompleteResult & ClickHouseAutocompleteResult {
```

Comment:
I don't think this is the correct type. Even though the one typed before is not correct either. It should be `A` (a type that inherits AutocompleteResultBase). But when I tried to fix it, it didn't work. 

Let's leave it as is, maybe? Your change doesn't bring anything new to the table, just makes it even more weirder, no?